src/maindx.cpp

```diff
@@ -1063,49 +1016,19 @@ extern "C" void ADDON_Destroy()
   initialized = false;
 }
 
-//-- GetStatus ---------------------------------------------------------------
-// Returns the current Status of this visualisation
-// !!! Add-on master function !!!
-//-----------------------------------------------------------------------------
-extern "C" ADDON_STATUS ADDON_GetStatus()
-{
-  return ADDON_STATUS_OK;
-}
-
 //-- SetSetting ---------------------------------------------------------------
 // Set a specific Setting value (called from XBMC)
 // !!! Add-on master function !!!
 //-----------------------------------------------------------------------------
-extern "C" ADDON_STATUS ADDON_SetSetting(const char *strSetting, const void* value)
+ADDON_STATUS CVisualizationShadertoy::SetSetting(const std::string& settingName, const kodi::CSettingValue& settingValue)
```

Comment:
By the use of `g_currentPreset = kodi::GetSettingInt("lastpresetidx");` and only the from addon self created value is this function no more really needed.